### PR TITLE
[ #6746 ] Using GHC 9.8.1 on stack.

### PR DESCRIPTION
**Pull request by asr**

### CODE REVIEW

**Comment by andreasabel** [comment]
Thanks!
To verify this build, it should be either added to https://github.com/agda/agda/blob/1354c8a76b7f2a4488f0e468623e2e9627b810a6/src/github/workflows/stack-dry-run.yml#L49
or become the official stack build version in https://github.com/agda/agda/blob/1354c8a76b7f2a4488f0e468623e2e9627b810a6/src/github/workflows/stack.yml#L14 https://github.com/agda/agda/blob/1354c8a76b7f2a4488f0e468623e2e9627b810a6/src/github/workflows/stack.yml#L55-L70

**Comment by asr** [comment]
Why didn't you add something similar for Cabal in #6746?

OK. I'll made the changes for verifying the build. Thanks!

File: stack-9.8.1.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+resolver: nightly-2023-10-17
```
Comment (andreasabel) [inline stack-9.8.1.yaml:1]: Nice trick!
`nightly` is officially still on GHC 9.6, but this seems to work!

File: stack-9.8.1.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+compiler: ghc-9.8.1
+compiler-check: match-exact
```
Comment (andreasabel) [inline stack-9.8.1.yaml:3]: How is that exact match working?  Shouldn't this trigger an error in the light of: https://github.com/commercialhaskell/stackage-snapshots/blob/bcb374ae9c949ec1332d3635bdf755e538d5cea4/nightly/2023/10/17.yaml#L13012

**Review by andreasabel — APPROVED** [review]

**Comment by andreasabel** [comment]
> Why didn't you add something similar for Cabal in #6746?

Oops, I forgot this, thanks for the reminder!

**Comment by andreasabel** [comment]
> > Why didn't you add something similar for Cabal in #6746?
> 
> Oops, I forgot this, thanks for the reminder!

Done in: https://github.com/agda/agda/pull/6924

File: stack-9.8.1.yaml (hunk 2 above)
Comment (andreasabel) [reply stack-9.8.1.yaml:3]: There is this issue "closed as completed":
- https://github.com/commercialhaskell/stack/issues/4282

Comment (andreasabel) [reply stack-9.8.1.yaml:3]: Ah, I think that `compiler:` was meant to match the compiler of the snapshot, but it seem to be the compiler used to build the present package.  I have always misunderstood this, it seems.

**Pull request merged**